storage: fix linters

tbg opened the pull request:
`./build/check-style.sh` looks like it's using `set -o pipefail`. However,
the actual behavior differed depending on whether one was using the `parallel`
or non-parallel branch in that script - the `parallel` invocation actually lost
the `set` options at the beginning of the script.

To make matters worse, many of the tests were unaware of the fact that they
should run with `-o pipefail` (and would report success if the main utility
they invoked failed when invoked with `-o pipefail`). Since CI runs with
`parallel`, this seems to have make CI work in practice, but when running
them sequentially, it would report success erroneously.

These scripts are pretty fragile. It would be good to make this less brittle
as it's burned a good amount of hours on this and I only noticed it by sheer
luck.

For example, I can't guarantee that I'm not introducing any other problems
with this change.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/9764)

<!-- Reviewable:end -->

jordanlewis commented on build/check-style.sh:
```diff
@@ -95,7 +95,7 @@ TestImportNames() {
 
 TestIneffassign() {
-  ! ineffassign . | grep -vF 'ineffectual assignment to sqlDollar' | grep -vF '.pb.go' # https://github.com/gogo/protobuf/issues/149
+  ! (ineffassign . || true) | grep -vF 'ineffectual assignment to sqlDollar' | grep -vF '.pb.go' # https://github.com/gogo/protobuf/issues/149
```
wait, doesn't this mean that all of these will immediately fail under pipefail? since we're returning 1 from the first part of the pipe.

tschottdorf replied:
The `!` binds weaker than the others. I parenthesized throughout to make this clearer.

tbg committed:
storage: fix linters

`./build/check-style.sh` looks like it's using `set -o pipefail`. However,
the actual behavior differed depending on whether one was using the `parallel`
or non-parallel branch in that script - the `parallel` invocation actually lost
the `set` options at the beginning of the script.

To make matters worse, many of the tests were unaware of the fact that they
should run with `-o pipefail` (and would report success if the main utility
they invoked failed when invoked with `-o pipefail`). Since CI runs with
`parallel`, this seems to have made CI work in practice, but when running
them sequentially, it would report success erroneously.

These scripts are pretty fragile. It would be good to make this less brittle as
it has burned a good amount of my hours and I only noticed it by sheer luck.

For example, I can't guarantee that I'm not introducing any other problems
with this change.

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9764#-:-KTLWNSQuc2sgnaNwv5K:bnfp4nl)*
<!-- Sent from Reviewable.io -->
